pkg/kubectl/run.go

```diff
@@ -847,23 +847,19 @@ func (BasicPod) Generate(genericParams map[string]interface{}) (runtime.Object,
 	return &pod, nil
 }
 
-// parseEnvs converts string into EnvVar objects.
+// parseEnvs converts a list of strings into EnvVar objects.
+// The strings should be of the form `key=value`.
 func parseEnvs(envArray []string) ([]v1.EnvVar, error) {
 	envs := make([]v1.EnvVar, 0, len(envArray))
 	for _, env := range envArray {
-		pos := strings.Index(env, "=")
-		if pos == -1 {
-			return nil, fmt.Errorf("invalid env: %v", env)
+		parts := strings.SplitN(env, "=", 2)
+		if len(parts) < 2 {
+			return nil, fmt.Errorf("multiple equals signs in EnvVar: %v", env)
```

Comment:
is that up-to-date? AFAIU, you would have less than two parts if you have NO equal sign in `EnvVar`.